cmd/admin-handlers.go

```diff
@@ -1882,3 +1885,91 @@ func (a adminAPIHandlers) ConsoleLogHandler(w http.ResponseWriter, r *http.Reque
 		}
 	}
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+
+	vars := mux.Vars(r)
+	switch infoType := vars["serviceType"]; infoType {
+	case "vault":
+		errorString := ""
+		if GlobalKMS == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
+			return
+		}
+		vaultInfo := madmin.VaultInfo{}
+
+		kmsEndPoint, name, authType, err := GlobalKMS.KMSInfo()
```

Comment:
I think we should stick to the idiom of returning early if we notice `err != nil` unless there is a good reason. In fact, returning the values configured via environment variables over the values present in the config.json must be internal to the `KMSInfo` method. In fact the value in `GlobalKMS` is filled with the above rule, so we don't need the environment variable handling code below.

Reply:
removed the code for envt variable .